### PR TITLE
Bug fix/cleanup 06052020

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

This fixes warnings such as
```
2020-05-05T23:35:15Z [24504] P WARNING [66666] {cluster} Loading the new plan took: 0.628109
``` 
that frequently occur in devel.

In current devel, we are currently spending lots of time in ClusterInfo::loadPlan(). The code in `loadPlan()` always throws away all `LogicalCollection` instances and replaces them with new ones. Even if the underlying collections didn't change. This permanent deletion and recreation of `LogicalCollection` instances is not free, and that's why in the past we added tuning knobs in `RocksDBCollection` and `RocksDBIndex` so that they don't create in-memory caches for these objects. More precise, we added an `isAStub()` getter to `LogicalCollection` that `RocksDBCollection` and `RocksDBIndex` check before adding an in-memory cache instance.

This PR reuses that logic so that we don't create revision tree objects for collections from the `ClusterInfo`. These are expensive to create, but unnecessary for the ClusterInfo's `LogicalCollection` object instances. 

This change should prevent the issue from happening in almost all cases.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *all tests*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9755/

### CODE REVIEW

File: lib/Rest/HttpRequest.cpp [hunk 1]
```diff
@@ -51,19 +51,6 @@ HttpRequest::HttpRequest(ConnectionInfo const& connectionInfo,
   _contentTypeResponse = ContentType::JSON;
 }
 
-// HACK HACK HACK
```
Comment (mchacki) [inline lib/Rest/HttpRequest.cpp:54]: good that this finally gone yay \o/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9757/

**Comment by jsteemann** [comment]
Tests blue

**Comment by jsteemann** [comment]
Re-running tests after small changes:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9765/

**Comment by jsteemann** [comment]
Tests ok

**Pull request merged**